### ISSUE TITLE
How to configure not to generate a pair before words or characters?

### ISSUE THREAD

**Issue by terroo** [jiangmiao/auto-pairs#302]
How to configure not to generate a pair before words or characters?
Example: 

[my word 
[99 number

**Comment by benfrain** [comment]
Yes, this was what I came here looking for. Imagine you want to wrap a string in template literals:

```
this.needsToBeWrapped
```

I move cursor to beginning:

```
|this.needsToBeWrapped
```

But as soon as I type the first opening curly brace I get this:

```
${}|this.needsToBeWrapped
```

When I want:

```
${|this.needsToBeWrapped
```

Could this be made to return out if current caret location is before an existing character?

**Comment by LunarWatcher** [comment]
Possible in my fork: https://github.com/LunarWatcher/auto-pairs/blob/master/doc/AutoPairs.txt#L606

**Comment by benfrain** [comment]
@LunarWatcher thanks. Looks like just what I needed.  I’ve just switched to nvim-autopairs as I’m on Lua but will be back to use yours if that doesn’t work out 🙏
